changelog/unreleased/pull-3419

```diff
@@ -0,0 +1,18 @@
+Enhancement: Use config file modes to derive new file/dir modes
+
+Previously restic would always create new files and directories on
+local/sftp repositories with hardcoded permissions of 0700/0600
+(rwx/rw). We now use the current read permissions of the "config" file to
+derive which modes new files and directories should be created with.
+
+For example the mapping in the cases below would look like this:
```

Comment:
Just state that a config file that is only readable by the user will lead to only user accessible files and that a group readable config will also allow the group to access the files. Btw. files are set to readonly once they are written, which makes the 0600 / 0660 permissions an implementation detail.

It might be worthwhile to explain how to switch a repository between both permission variants.